src/Microsoft.Extensions.Logging.Log4Net.AspNetCore/Log4NetLogger.cs

```diff
@@ -97,24 +100,54 @@ public bool IsEnabled(LogLevel logLevel)
                 return;
             }
 
-            var message = PrepareMessage(logLevel, eventId, state, exception, formatter);
+            var message = PrepareMessage(logLevel, eventId, state, exception, formatter, ScopeProvider);
             LogMessage(message);
         }
 
         private void LogMessage<TState>(MessageCandidate<TState> candidate)
         {
             LoggingEvent loggingEvent = options.LoggingEventFactory.CreateLoggingEvent(candidate, logger, options);
-
+            
             if (loggingEvent == null)
                 return;
 
             this.logger.Log(loggingEvent);
         }
 
-        private static MessageCandidate<TState> PrepareMessage<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        private static MessageCandidate<TState> PrepareMessage<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter, IExternalScopeProvider externalScopeProvider)
         {
             EnsureValidFormatter(formatter);
-            return new MessageCandidate<TState>(logLevel, eventId, state, exception, formatter);
+
+            var scopes = new List<ScopeInfo>(10);
+
+            // gather info about scope(s), if any
+            if (externalScopeProvider != null)
```

Comment:
I'm not the maintainer so you should wait for other feedback before changing anything, but:

I feel like this reduced configuratiblity a lot. Before you could pretty much do anything with the scope if you just provided the right ScopeFactory. And now you are limited to the two casts and that's it.

I would rather see this processing done in the LoggingEvent Factory so someone could plug their own implementation.
Maybe to reduce allocations this even could be done by passing the ExternalScopeProvider to the LogginEventFactory directly.

At least that's how it's done for the console formatters in the runtime https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleFormatter.cs#L35.

That might result in a breaking change, but I implemented this feature a few weeks ago so I don't suspect to many people are using it.